### PR TITLE
Replace NonNegativeLengthOrNumber by a specific type for -moz-tab-size

### PR DESCRIPTION
**Pull request by nox**
This is the only use of this type.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/20131)

### CODE REVIEW

File: components/style/values/specified/text.rs [hunk 1]
```diff
@@ -514,3 +515,18 @@ impl ToComputedValue for TextAlign {
...
+
+/// A specified value for the `-moz-tab-size` property.
+pub type MozTabSize = GenericMozTabSize<NonNegativeNumber, NonNegativeLength>;
+
+impl Parse for MozTabSize {
+    fn parse<'i, 't>(
+        context: &ParserContext,
+        input: &mut Parser<'i, 't>,
+    ) -> Result<Self, ParseError<'i>> {
+        if let Ok(number) = input.try(|i| NonNegativeNumber::parse(context, i)) {
```
Comment (emilio) [inline components/style/values/specified/text.rs]: Maybe worth still pointing out that it's important that `NonNegativeNumber` is parsed before `NonNegativeLength`?

**Review by emilio — APPROVED** [review]

File: components/style/values/specified/text.rs (hunk 1 above)
Comment (nox) [reply components/style/values/specified/text.rs]: Done.

**Pull request merged**